comp/otelcol/converter/impl/prometheus.go

```diff
@@ -0,0 +1,195 @@
+// which points to the collectors internal telemetry metrics. In cases where this is not true, it adds
+// a pipeline with the prometheus exporter and datadog exporter.
+// todo(mackjmr): in the case where there are two datadog exporters with the same API key, we may not
+// want to configure two pipelines with the prometheus receiver for each exporter, as this will lead
+// to shipping the health metrics twice to the same org. If there are two datadog exporters with
+// different API keys, there is no way to know if these are from the same org, so there is a risk
+// of double shipping.
+func addPrometheusReceiver(conf *confmap.Conf, comp component) {
+	datadogExportersMap := getDatadogExporters(conf)
+	internalMetricsAddress := conf.Get("service::telemetry::metrics::address")
+	if internalMetricsAddress == nil {
+		internalMetricsAddress = "0.0.0.0:8888"
+	}
+	stringMapConf := conf.ToStringMap()
+
+	// find prometheus receivers which point to internal telemetry metrics. If present, check if it is defined
+	// in pipeline with DD exporter. If so, remove from datadog exporter map.
+	if receivers, ok := stringMapConf["receivers"]; ok {
+		if receiversMap, ok := receivers.(map[string]any); ok {
+			for receiver, receiverConfig := range receiversMap {
+				if componentName(receiver) == comp.Name {
+					if receiverConfigMap, ok := receiverConfig.(map[string]any); ok {
+						if prometheusConfig, ok := receiverConfigMap["config"]; ok {
+							if prometheusConfigMap, ok := prometheusConfig.(map[string]any); ok {
+								if prometheusScrapeConfigs, ok := prometheusConfigMap["scrape_configs"]; ok {
+									if prometheusScrapeConfigsSlice, ok := prometheusScrapeConfigs.([]any); ok {
+										for _, scrapeConfig := range prometheusScrapeConfigsSlice {
+											if scrapeConfigMap, ok := scrapeConfig.(map[string]any); ok {
+												if staticConfig, ok := scrapeConfigMap["static_configs"]; ok {
+													if staticConfigSlice, ok := staticConfig.([]any); ok {
+														for _, staticConfig := range staticConfigSlice {
+															if staticConfigMap, ok := staticConfig.(map[string]any); ok {
+																if targets, ok := staticConfigMap["targets"]; ok {
+																	if targetsSlice, ok := targets.([]any); ok {
+																		for _, target := range targetsSlice {
+																			if targetString, ok := target.(string); ok {
+																				if targetString == internalMetricsAddress {
+																					if ddExporter := receiverInPipelineWithDatadogExporter(conf, receiver); ddExporter != "" {
+																						delete(datadogExportersMap, ddExporter)
+																					}
+																				}
+																			}
+																		}
+																	}
+																}
+															}
+														}
+													}
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
```

Comment:
Avoid nested loops, by returning if `!ok` .  That is preferred golang style.

Reply:
Replaced by `return` or `continue`. I think this makes the code a bit easier to understand, but there are a lot of:
```
if !ok {
    return
}
```
```
if !ok {
    continue
}
```

Reply:
That is fine. and is preferred golang style.